lib/PaymentService.js

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) 2018-2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import axios from 'axios';
+
+/**
+ * Wrapper around the bedrock-payment-http API.
+ *
+ * @param {object} [options] - Options to use.
+ * @param {string} [options.baseURL = '/payment'] - The baseURL for
+ *   the payment api.
+ * @param {object} [options.httpsAgent = axios] - The module used to
+ *   fetch from the api.
+ *
+ * @returns {PaymentService} The PaymentService.
+ */
+export default class PaymentService {
+  constructor({baseURL = '/payment', httpsAgent = axios} = {}) {
+    this.baseURL = baseURL;
+    this.httpsAgent = httpsAgent;
```

Comment:
I'm concerned that the use of `httpsAgent` here does not seem consistent with how that term is used elsewhere, for instance: https://github.com/digitalbazaar/web-ledger-client/blob/master/lib/WebLedgerClient.js#L31-L32

Also, seems to me that services need to define some good defaults for axios which would include setting a timeout value instead of the silly default of no timeout.

Reply:
I think it might be better to make this module more light weight and remove axios entirely and just throw if no httpsAgent is there. That said the name for httpsAgent might be confusing with back end code, but I thought making the module used to make the actual api call configurable might be better in the future.

Reply:
Well, this points to a larger issue in the front end, which is that if services like this do not abstract away the interface and errors produced by a particular xhr request library, then then there is really no ability to swap out one module for another. I know that you made an attempt to address the error handling in the past, but that is still an unsolved problem today.

Reply:
so what you are saying is that as long as we can not get consistent error messages across multiple modules we have a problem. We could decide on any library that implements the fetch api: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

There are plenty of existing npm projects that implement it.